flink-runtime/src/main/java/org/apache/flink/runtime/topology/BaseTopology.java

```diff
@@ -30,7 +30,7 @@
 	 *
 	 * @return topologically sorted iterable over all vertices
 	 */
-	Iterable<V> getVertices();
+	Iterable<? extends V> getVertices();
```

Comment:
I think we need to do similar changes for:
`PipelinedRegion#getVertices()`
`PipelinedRegion#getConsumedResults()`
`Topology#getAllPipelinedRegions()`

Reply:
Right.